src/TheCompact.sol

```diff
@@ -42,82 +42,59 @@ contract TheCompact is ITheCompact, ERC6909, TheCompactLogic {
     using EfficiencyLib for bool;
     using EfficiencyLib for uint256;
 
-    function deposit(address allocator) external payable returns (uint256) {
```

Comment:
do you think it makes sense to drop these default options? simple deposits will actually end up taking *more* calldata after this change